salt/utils/process.py

```diff
@@ -55,7 +56,17 @@ def notify_systemd():
         import systemd.daemon
     except ImportError:
         if salt.utils.which('systemd-notify') and systemd_notify_call('--booted'):
-            return systemd_notify_call('--ready')
+            # Notify systemd synchronously
+            notify_socket = os.getenv('NOTIFY_SOCKET')
+            if notify_socket:
+                # Handle abstract namespace socket
+                if notify_socket.startswith('@'):
+                    notify_socket = '\0{0}'.format(notify_socket[1:])
+                sock = socket.socket(socket.AF_UNIX, socket.SOCK_DGRAM)
+                sock.connect(notify_socket)
+                sock.sendall('READY=1'.encode())
+                sock.close()
+                return True
```

Comment:
@evverx: Thanks for your comment, I pushed another commit to address your concerns. Let me know if it doesn't look a expected, I was not sure how to cause the `socket.error` to happen in order to actually test this.